sdk/core/Azure.Core/src/Pipeline/LoggingPolicy.cs

```diff
@@ -132,9 +140,15 @@ private string FormatHeaders(IEnumerable<HttpHeader> headers)
             var stringBuilder = new StringBuilder();
             foreach (HttpHeader header in headers)
             {
-                if (_allowedHeaderNames.Contains(header.Name))
+                stringBuilder.Append(header.Name);
+                stringBuilder.Append(':');
+                if (_logAllHeaders || _allowedHeaderNames.Contains(header.Name))
+                {
+                    stringBuilder.AppendLine(header.Value);
+                }
+                else
                 {
-                    stringBuilder.AppendLine(header.ToString());
+                    stringBuilder.AppendLine("*");
```

Comment:
Would it be helpful to have something that indicates at a glance that we redacted this to avoid confusion?  For (an obvious) example, something like `<< REDACTED >>` ?

Reply:
I thought about it and I'm not sure. I don't like how long things get when `<< REDACTED >>` is used.

Reply:
`*` is a valid value for things like `If-Match` though.  Storage has historically used the string `SANITIZED` for the two pieces of data it deny-listed.

Reply:
A fair objection.   My concern is just that `*` is brief and anonymous enough that someone not familiar with the implementation troubleshooting via logs may mistakenly believe that value was actually used for the header.

Reply:
Okay, guideline use word `redact` so I think `REDACTED` (without `<< >>`) would be a middle ground.